server/src/main/java/io/spine/server/entity/storage/Columns.java

```diff
@@ -147,94 +107,158 @@ public static void checkColumnDefinitions(EntityClass<?> entityClass) {
+
+        return entityColumns;
     }
 
     /**
-     * Generates the {@link EntityColumn} values considering the passed
-     * {@linkplain Entity entity type} indexed.
+     * Generates the {@linkplain EntityColumn column} values for the given {@linkplain Entity}.
+     *
+     * <p>Retrieves {@linkplain EntityColumn columns} for the given {@code Entity} class, then generates
+     * {@linkplain MemoizedValue memoized values} from them.
      *
-     * @param entityType indexed type of the {@link Entity}
-     * @param entity     the object which to take the values from
-     * @return a {@link Map} of the {@linkplain EntityColumn columns}
+     * @param entity an {@link Entity} to get the {@linkplain EntityColumn column} values from
+     * @param <E>    the type of the {@link Entity}
+     * @return a {@link Map} of the column {@linkplain EntityColumn#getStoredName()
+     *         names for storing} to their {@linkplain MemoizedValue memoized values}
+     * @see MemoizedValue
      */
-    private static Map<String, MemoizedValue> extractColumns(Class<? extends Entity> entityType,
-                                                             Entity entity) {
-        final Collection<EntityColumn> columns = knownEntityProperties.get(entityType);
-        final Map<String, MemoizedValue> values = new HashMap<>(columns.size());
+    static <E extends Entity<?, ?>> Map<String, EntityColumn.MemoizedValue> extractColumnValues(E entity) {
+        checkNotNull(entity);
 
-        for (EntityColumn column : columns) {
-            final String name = column.getStoredName();
-            final MemoizedValue value = column.memoizeFor(entity);
-            values.put(name, value);
-        }
-        return values;
+        final Collection<EntityColumn> entityColumns = obtainColumns(entity.getClass());
+        return extractColumnValues(entity, entityColumns);
     }
 
-    private static void ensureRegistered(Class<? extends Entity> entityType) {
-        if (knownEntityProperties.containsKey(entityType)) {
-            return;
+    /**
+     * Generates the {@linkplain EntityColumn column} values for the given {@linkplain Entity}.
+     *
+     * <p>Uses given {@linkplain EntityColumn entity columns} for the value extraction.
+     *
+     * <p>This way the process of {@linkplain Columns#obtainColumns(Class) obtaining columns} from
+     * the given {@link Entity} class can be skipped.
+     *
+     * @param entity        an {@link Entity} to get the {@linkplain EntityColumn column} values from
+     * @param entityColumns {@linkplain EntityColumn entity columns} which values should be extracted
+     * @param <E>           the type of the {@link Entity}
+     * @return a {@link Map} of the column {@linkplain EntityColumn#getStoredName()
+     *         names for storing} to their {@linkplain MemoizedValue memoized values}
+     * @see MemoizedValue
+     */
+    static <E extends Entity<?, ?>> Map<String, MemoizedValue> extractColumnValues(
```

Comment:
I think the documentation of this method lacks the fact that we don't process non-`public` classes (as I can tell from the implementation, of course; but still it's an important thing for those who judges by the docs).